src/pmdas/bpf/modules/netatop.c

```diff
@@ -0,0 +1,473 @@
+/*
+ * Based on netatop-bpf:
+ * https://github.com/bytedance/netatop-bpf
+ */
+
+#include "module.h"
+
+#include <assert.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <signal.h>
+#include <string.h>
+#include <errno.h>
+#include <dirent.h>
+#include <sys/resource.h>
+#include <bpf/libbpf.h>
+#include <bpf/bpf.h>
+#include <pcp/pmda.h>
+#include <sys/queue.h>
+#include <pcp/pmwebapi.h>
+#include <string.h>
+
+#include "netatop.skel.h"
+#include "libpcp.h"
+#include "netatop.h"
+#include "btf_helpers.h"
+
+#define INDOM_COUNT 1
+#define MAX_PORTS 64
+
+static struct env {
+    int process_count;
+} env = {
+    .process_count = 20,
+};
+
+int tgid_map_fd;
+int nr_cpus;
+
+static pmdaInstid *netatop_instances = NULL;
```

Comment:
No need to assign to NULL here, its mandatory for the compiler to do this for globals.